awx/main/tasks.py

```diff
@@ -867,6 +869,24 @@ def run(self, pk, isolated_host=None, **kwargs):
                 status, rc = manager_instance.run(instance, isolated_host,
                                                   kwargs['private_data_dir'],
                                                   kwargs.get('proot_temp_dir'))
+            elif hasattr(self, 'management_command'):
+                try:
+                    old_env = os.environ.copy()
```

Comment:
I wrote a context manager in 3.2 (for a different feature) that handles this for you:

https://github.com/ansible/awx/blob/devel/awx/main/utils/common.py#L675